Classes/Service/DeeplGlossaryService.php

```diff
@@ -24,14 +26,18 @@ final class DeeplGlossaryService
 
     protected GlossaryRepository $glossaryRepository;
 
+    protected EventDispatcher $eventDispatcher;
+
     public function __construct(
         FrontendInterface $cache,
         ClientInterface $client,
-        GlossaryRepository $glossaryRepository
+        GlossaryRepository $glossaryRepository,
+        EventDispatcher $eventDispatcher,
```

Comment:
```suggestion
        EventDispatcherInterface $eventDispatcher,
```

It's common practice to use the `EventDispatcherInterface` as type declaration and not the concrete implementation. We strive to use interfaces over concrete implementation as much as possible, and refactor the code base over time if not
breaking over time - and if we have the time for it. 

For new code we are having a eye on such things.

Reply:
Changed it